### PR TITLE
[Datasets] Add missing passthrough args to read_images()

### LINKED ISSUE #32941: [Datasets] Imagery reading doesn't work with `local://` URIs
This PR added a `local_uri` arg that's eventually passed to file-based datasource's `_read_file()` and `_read_stream()` APIs; this was only needed for the `ParquetDatasource`, where all other datasources ignore it in their `**reader_args` catch-all (for example: [`CSVDatasource`](https://github.com/ray-project/ray/blob/a892241ca7574af47f278a667e6493a4b03686d7/python/ray/data/datasource/csv_datasource.py#L30)). Unfortunately, `ImageDatasource` [doesn't have the `**reader_args` catch-all](https://github.com/ray-project/ray/blob/a892241ca7574af47f278a667e6493a4b03686d7/python/ray/data/datasource/image_datasource.py#L75-L81), which causes this `local_uri` arg pass to fail. This wasn't caught because the image datasource doesn't have full local/remote filesystem coverage like the tabular API test suites.

We should add that catch-all to `ImageDatasource._read_file()` and add local + remote filesystem coverage for both the image datasource and the binary datasource.

### PR DESCRIPTION
**Pull request by clarkzinzow**
This PR adds missing passthrough args to `read_images()`, such as `ray_remote_args`, `arrow_open_file_args`, and other misc. args popped in the base `FileBasedDatasource` such as `compression`. This PR also adds a `**read_args` catch-all to `ImageDatasource._read_file()`, which should add support for using the `local://` protocol.

## Related issue number

Closes https://github.com/ray-project/ray/issues/32941

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by jianoaix — APPROVED** [review]

**Review by krfricke — APPROVED** [review]
Thanks!

**Review by c21 — APPROVED** [review]
LG assuming unit test to be added.

File: python/ray/data/read_api.py [hunk 1]
```diff
@@ -628,6 +630,9 @@ def read_images(
         filesystem: The filesystem implementation to read from.
         parallelism: The requested parallelism of the read. Parallelism may be
             limited by the number of files of the dataset.
+        ray_remote_args: kwargs passed to ray.remote in the read tasks.
+        arrow_open_file_args: kwargs passed to
+            ``pyarrow.fs.FileSystem.open_input_file``.
```
Comment (c21) [inline python/ray/data/read_api.py:635]: nit: can we follow the same ordering, to add these two parameters after `meta_provider`?

Comment (clarkzinzow) [reply python/ray/data/read_api.py:635]: Ah yep, looks like these got reordered when resolving merge conflicts, I'll fix that!

**Review by c21 — APPROVED** [review]

**Review by zhe-thoughts — APPROVED** [review]
Approved for merging to master

**Pull request merged**